### PR TITLE
Upload packages along with AWS advanced templates

### PR DESCRIPTION
**Pull request by branden**
## High Level Description

This includes packages in the artifacts that get uploaded along with generated AWS templates.

This also fixes an issue where failed downloads to tempfile locations during node setup cause an incidental error that hides the original download error.

## Related Issues

  - [DCOS-14381](https://jira.mesosphere.com/browse/DCOS-14381) Various reports of failure when using templates generated by --aws-cloudformation

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: We need an integration test covering advanced template gen + deploy, which is outside the scope of this PR.
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
Ready for review @amitaekbote @orsenthil.

File: pkgpanda/__init__.py [hunk 1]
```diff
@@ -302,7 +302,7 @@ def requests_fetcher(base_url, id_str, target, work_dir):
     # TODO(cmaloney): Use a private tmp directory so there is no chance of a user
     # intercepting the tarball + other validation data locally.
     with tempfile.NamedTemporaryFile(suffix=".tar.xz") as file:
-        download(file.name, url, work_dir)
+        download(file.name, url, work_dir, rm_on_error=False)
```
Comment (amitaekbote) [inline pkgpanda/__init__.py]: Why dont we want to remove on error?

Comment (branden) [inline pkgpanda/__init__.py]: > If download() fails and cleans up a tempfile, the error it encountered will be lost when the tempfile tries and fails to rm itself.

Comment (amitaekbote) [inline pkgpanda/__init__.py]: I missed the NamedTemporaryFile which will clean up itself
https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

**Pull request merged**